src/ReverseProxy/Service/Management/ProxyConfigManager.cs

```diff
@@ -156,8 +214,9 @@ private async Task ApplyConfigAsync(IProxyConfig config)
             }
 
             // Update clusters first because routes need to reference them.
-            UpdateRuntimeClusters(configuredClusters);
-            UpdateRuntimeRoutes(configuredRoutes);
+            var clustersChanged = UpdateRuntimeClusters(configuredClusters);
+            var routesChanged = UpdateRuntimeRoutes(configuredRoutes);
+            return routesChanged || clustersChanged;
```

Comment:
```suggestion
            return routesChanged;
```
Now that cluster changes are propagated on route changes with revision, I think it's possible to only rely on `routesChanged` to recreate endpoints. 

There's no need to recreate the endpoints if an unused cluster is added, edited or removed, right?

Reply:
I think you're right. The strangest change I can think of is if an in-use cluster were removed, but rebuilding routes in that case doesn't seem helpful.